lib/tasks/events.rake

```diff
@@ -0,0 +1,187 @@
+    [] Give the user an option to pull student data from course id
+=end
+
+def main(users, asset_types, context_id)
+  prompt_user :delete_events
+  num_records = prompt_user :num_records
+  chunk_size = prompt_user :chunk_size
+  start_date = prompt_user :start_date
+  end_date = prompt_user :end_date
+
+  create_rand_event_records(
+    num_records,
+    users,
+    asset_types,
+    context_id,
+    chunk_size,
+    start_date,
+    end_date,
+  )
+end
+
+def prompt_user option
+  case option
+  when :delete_events
+    puts "Would you like to delete all previous events? [yes/no]"
+    user_choice = STDIN.gets.chomp
+
+    until ["yes", "no"].include?(user_choice)
+      puts "Invalid option: \"#{user_choice}\" - please input yes or no"
+      user_choice = STDIN.gets.chomp
+    end
+
+    unless user_choice == "no"
+      puts "Deleting all records from canvas_events table"
+      CanvasEvent.delete_all
+    end
+
+  when :num_records
+    puts "How many event records would you like to create?"
+    num_records = STDIN.gets.chomp.to_i
+
+    until num_records > 0
+      puts "Please input a number greater than 0"
+      num_records = STDIN.gets.chomp.to_i
+    end
+
+    num_records
+
+  when :chunk_size
+    puts "How many random events do you want to assign to a student per iteration?"
+    puts "[chunk_size/info]"
+    user_choice = STDIN.gets.chomp
+
+    if user_choice == "info"
+      puts "For example if you would like to create 1000 records the number"
+      puts "you input here (n) will be used to assign a student 1..n random"
+      puts "live events. Let\'s say rand(n = 50) + 1 = 50. 50.times random records"
+      puts "will be created for a random student. That\'s one iteration. The"
+      puts "remaining records will now be 950."
+      puts "Make sense? Ok, go ahead and enter a number"
+      user_choice = STDIN.gets.chomp
+    end
+
+    until user_choice.to_i > 0
+      puts "Please input a number greater than 0"
+      user_choice = STDIN.gets.chomp.to_i
+    end
+
+  when :start_date
+    puts "Please specify a start date [yyyy-mm-dd]"
+    user_input = STDIN.gets.chomp
+    start_date = nil
+
+    until start_date.class.to_s == "DateTime"
+      begin
+        start_date = user_input.to_datetime
+      rescue ArgumentError
+        puts "Sorry, that format can\'t be read. Please input a start date using yyyy-mm-dd"
+        user_input = STDIN.gets.chomp
+      end
+    end
+
+    start_date.utc.iso8601
+
+  when :end_date
+    puts "Please specify an end date [yyyy-mm-dd]"
+    user_input = STDIN.gets.chomp
+    end_date = nil
+
+    until end_date.class.to_s == "DateTime"
+      begin
+        end_date = user_input.to_datetime
+      rescue ArgumentError
+        puts "Sorry, that format can\'t be read. Please input a end date using yyyy-mm-dd"
+        user_input = STDIN.gets.chomp
+      end
+    end
+
+    end_date.utc.iso8601
+  end
+end
+
+def time_rand(start_date, end_date)
```

Comment:
This could probably be replaced with `rand(start_date..end_date.end_of_day).iso8601`